tempest/api/workloadmgr/cli/test_wlm_cli.py

```diff
@@ -276,3 +276,221 @@ def test_05_create_workload_setting_with_invalid_values(self):
+                LOG.debug("Command executed correctly")
+            else:
+                reporting.add_test_step(
+                    "Verify correct error thrown for invalid parameter", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+            # Cleanup
+            # Delete workload setting
+            wl_setting_delete = command_argument_string.workload_setting_delete + \
+                                workload_setting_name
+            LOG.debug("Workload setting delete command: {}".format(wl_setting_delete))
+            rc = cli_parser.cli_returncode(wl_setting_delete)
+            time.sleep(10)
+            wc = query_data.get_created_workload_setting(workload_setting_name)
+            LOG.debug("Workload setting status: " + str(wc))
+
+            if wc:
+                reporting.add_test_step(
+                    "workload setting deletion failed", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+            else:
+                reporting.add_test_step(
+                    "workload setting deleted successfully", tvaultconf.PASS)
+
+        except Exception as e:
+            LOG.error("Exception: " + str(e))
+            reporting.set_test_script_status(tvaultconf.FAIL)
+        finally:
+            reporting.test_case_to_write()
+
+    @decorators.attr(type='workloadmgr_cli')
+    def test_08_workload_setting_list(self):
+        reporting.add_test_script(str(__name__) + "_create_workload_setting_cli_list")
+        try:
+            # workload setting list with CLI command
+            wl_setting_list = command_argument_string.workload_setting_list + " --get_hidden True -f value | wc -l"
+            LOG.debug("Workload setting list command: {}".format(wl_setting_list))
+            rc = cli_parser.cli_returncode(wl_setting_list)
+            if rc == 0:
+                reporting.add_test_step(
+                    "Execute workload setting-list", tvaultconf.PASS)
+                LOG.debug("workload setting-list command executed correctly")
+            else:
+                reporting.add_test_step(
+                    "Execute workload setting-list", tvaultconf.FAIL)
+                raise Exception("workload setting-list command did not execute correctly")
+
+            out = cli_parser.cli_output(wl_setting_list)
+            LOG.debug("Response from CLI: " + str(out))
+
+            # Compare values with database for workloadmgr setting-list
+            wc = query_data.get_db_rows_count("settings", "hidden", str(1))
+            LOG.debug("workload settings in db: {}".format(wc))
+            if (wc == (int(out) - 4)):
+                reporting.add_test_step(
+                    "Verify workload setting list with DB", tvaultconf.PASS)
+            else:
+                reporting.add_test_step(
+                    "Verify workload setting list with DB", tvaultconf.FAIL)
```

Comment:
In case of failure, do we need to mark this case as failed ?

Reply:
If the workload setting verification with DB fails, the test should fail, as there is discrepancy between cli result and DB.